src/NewProject/location_form.cpp

```diff
@@ -23,10 +22,17 @@ locationForm::locationForm(QWidget *parent)
   ui->m_checkBox->setText(tr("Create Project Subdirectory"));
   ui->m_labelPath0->setText(tr("Project will be created at:"));
   ui->m_btnBrowse->setText(tr("Browse..."));
-  ui->m_lineEditPname->setText("project_1");
+  ui->m_lineEditPname->setText(QString("project_") + QString::number(count));
   ui->m_lineEditPpath->setText(QDir::currentPath());
   ui->m_labelPath1->setText(ui->m_lineEditPpath->text());
   ui->m_checkBox->setCheckState(Qt::CheckState::Checked);
+
+  QString name = ui->m_lineEditPname->text();
```

Comment:
why not:

QString name = QString("project_") + QString::number(count);
while (QDir(name).exists()) {
    name = QString("project_") + QString::number(count++);
  };
ui->m_lineEditPname->setText(name);

Reply:
> Hey @zaryabkhan-rs, I had the same comments as Volodymyr. To provide some context for those comments, count is used only one time for determining the project number so a class variable is overkill here; A simple local variable in that function will do fine. For your while loop, you have the correct logic, but each time you assign the value to the UI field which will be done again if the value needs to increment. As such, you can just increment your counter in the loop until you get a valid file name and then you can assign that string to the ui field _once_ afterwards.
> 
> For example, I solved the same problem by putting the following at the top of the function:
> 
> ```
>   int count = 1;
>   QString project_prefix = "project_";
>   while (QDir(project_prefix + QString::number(count)).exists()) {
>     count++;  
>   }
> ```
> 
> which increments count until project_{count} is an unused project name
> 
> and then in the original call to set the field's value i did:
> 
> ```
> ui->m_lineEditPname->setText(project_prefix + QString::number(count));
> ```
> 
> Note that this also somewhat addresses your question about "project_" as a default. I think this is fine since we just need something default that the user can change if they want, but by defining project_prefix and using that variable in our logic, it leaves the code in a state where if we ever did decide to change "project_" we only need to update the line where we set project_prefix.

Yeah! you are right! I have modified it. Now it looks more consistent. Thankyou for suggestion.